### PR TITLE
MM 45015 - auto follow threads

### PR DESCRIPTION
**Pull request by tanmaythole**
#### Summary

Added auto follow thread select option

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/22386
https://mattermost.atlassian.net/browse/MM-45015

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file updates
- [x] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on:  Samsung M11 - 12 version

#### Release Note

```release-note
Added a new select option to auto-follow all threads in the Channel's Info.
```

### CODE REVIEW

File: app/screens/channel_info/options/index.tsx [hunk 1]
```diff
@@ -32,9 +33,12 @@ const Options = ({
 
     return (
         <>
-            {type !== General.DM_CHANNEL &&
-                <IgnoreMentions channelId={channelId}/>
-            }
+            {type !== General.DM_CHANNEL && (
+                <>
+                    <AutoFollowThreads channelId={channelId}/>
```
Comment (enahum) [inline app/screens/channel_info/options/index.tsx:38]: this should only be available if the user has CRT enabled, isn't it?

File: app/screens/channel_info/options/auto_follow_threads/auto_follow_threads.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+import {useIntl} from 'react-intl';
+
+import {updateChannelNotifyProps} from '@actions/remote/channel';
+import OptionItem from '@app/components/option_item';
+import {useServerUrl} from '@app/context/server';
```
Comment (enahum) [inline app/screens/channel_info/options/auto_follow_threads/auto_follow_threads.tsx:9]:
```suggestion
import OptionItem from '@components/option_item';
import {useServerUrl} from '@context/server';
```

**Review by enahum — CHANGES_REQUESTED** [review]

**Comment by larkox** [comment]
@tanmaythole Are you still working on this? Could you address the comments?

**Comment by tanmaythole** [comment]
> @tanmaythole Are you still working on this? Could you address the comments?

Yes, give me some time. I will commit changes as soon as possible.

File: app/screens/channel_info/options/auto_follow_threads/auto_follow_threads.tsx (hunk 2 above)
Comment (tanmaythole) [reply app/screens/channel_info/options/auto_follow_threads/auto_follow_threads.tsx]: Done 👍🏻

File: app/screens/channel_info/options/index.tsx [hunk 3]
```diff
@@ -20,21 +21,26 @@ type Props = {
...
     canManageMembers,
+    isCRTEnabled,
 }: Props) => {
     const isDMorGM = isTypeDMorGM(type);
 
     return (
         <>
-            {type !== General.DM_CHANNEL &&
+            {isCRTEnabled && (
+                <AutoFollowThreads channelId={channelId}/>
```
Comment (enahum) [inline app/screens/channel_info/options/index.tsx:39]: does this also apply to DM's ?

File: app/screens/channel_info/options/auto_follow_threads/auto_follow_threads.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+};
+
+const AutoFollowThreads = ({channelId, followedStatus}: Props) => {
+    const [autoFollow, setAutoFollow] = useState(followedStatus);
+    const serverUrl = useServerUrl();
+    const {formatMessage} = useIntl();
+
+    const toggleFollow = preventDoubleTap(() => {
+        const props: Partial<ChannelNotifyProps> = {
+            channel_auto_follow_threads: followedStatus ? 'off' : 'on',
```
Comment (enahum) [inline app/screens/channel_info/options/auto_follow_threads/auto_follow_threads.tsx:24]: you should be using the constants declared in the constants/channel file

File: app/screens/channel_info/options/auto_follow_threads/auto_follow_threads.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        };
+        setAutoFollow(!autoFollow);
+        updateChannelNotifyProps(serverUrl, channelId, props);
```
Comment (enahum) [inline app/screens/channel_info/options/auto_follow_threads/auto_follow_threads.tsx:27]: what happens if this fails? I know is the same with other options, just want to see what would be the consequence and if we should also thing about this for the other options as well.

**Review by enahum — APPROVED** [review]

**Review by enahum — CHANGES_REQUESTED** [review]
I think you need to run `i18n-extract` as is missing the `en.json` file

**Review by larkox — APPROVED** [review]
No extra comments from my side.

File: app/screens/channel_info/options/auto_follow_threads/auto_follow_threads.tsx (hunk 5 above)
Comment (tanmaythole) [reply app/screens/channel_info/options/auto_follow_threads/auto_follow_threads.tsx:27]: Okay need to add checks for this and other option as well

**Review by lindy65 — APPROVED** [review]
Thanks @tanmaythole 

I'm going to `qa-defer` this PR as I'm having trouble accessing `master.test.mattermost.com` from mobile.  I will test post merge.

cc// @johndavidlugtu

**Comment by hanzei** [comment]
@larkox @enahum Can we merge this PR?

**Pull request merged**